project-koku/koku
# COST-3317: missing cost_category for monthly cost-model derived rows

## Jira Ticket

[COST-3317](https://issues.redhat.com/browse/COST-3317)

## Description

This change will add cost_category_id to the monthly rows inserted for the cost model.

## Testing


----


We can use iqe testing branch. 
First we are going to test that this will fix rows that have been inserted without a cost_category_id retroactively:

1. Checkout main
2. Add a breakpoint to the test
Open iqe and go to the `test_api_ocp_all_bucketing` test. Add a breakpoint after the `call_api()`.

3. Run the smoke test
Run the test with:
```
ENV_FOR_DYNACONF=local iqe tests plugin cost_management -k 'test_api_ocp_all_bucketing'  --pdb
```
4. When you hit your breakpoint check the database to see that rows were entered without a `cost_category_id`
```
postgres=# select * from reporting_ocpusagelineitem_daily_summary WHERE cost_category_id IS NULL and namespace LIKE '%openshift-%';
-[ RECORD 1 ]----------------------------------+-----------------------------------------------
uuid                                           | 61464d3d-c6e2-4a93-8d37-a17b6f2d6ba9
cluster_id                                     | test_cost_ocp_on_aws_cluster
cluster_alias                                  | test_cost_ocp_on_aws_cluster
data_source                                    | Pod
namespace                                      | openshift-stage
node                                           | ocp-nod
resource######################################
infrastructure_project_monthly_cost            | {"cpu": 500.0000000000, "pvc": 0, "memory": 0}
supplementary_project_monthly_cost             | 
monthly_cost_type                              | Node
source_uuid                                    | 5f2661a7-c839-4181-bc99-e8eefc09ee75
report_period_id                               | 10
raw_currency                                   | 
cost_category_id                               | 
``` 
5. Switch to this branch (`git checkout cost-3317-cost-category-monthly-rates`)
6. Hit the following endpoints to rerun the cost model updates:
you can find your source_uuid by going to following endpoint:
`ctrl + f source_uuid`
```
http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=*&filter[category]=*&start_date=2022-11-01&end_date=2022-12-07&category=platform
```

Previous month 
a. `http://127.0.0.1:5042/api/cost-management/v1/update_cost_model_costs/?provider_uuid=31f007f9-c760-43e9-a118-dcdba7f42f71&schema=org1234567&start_date=2022-11-01&end_date=2022-11-30`

Current Month:
b. http://127.0.0.1:5042/api/cost-management/v1/update_cost_model_costs/?provider_uuid=31f007f9-c760-43e9-a118-dcdba7f42f71&schema=org1234567&start_date=2022-12-01&end_date=2022-12-30

7. Recheck the database:
```
postgres=# select * from reporting_ocpusagelineitem_daily_summary WHERE cost_category_id IS NULL and namespace LIKE '%openshift-%';
(0 rows)
```
8. Continue the iqe test now that we have updated the cost model with the new code:
```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================= 4 passed, 10263 deselected, 70 warnings in 193.67s (0:03:13) =============================================
```


## Notes

...
